core/src/main/java/com/linecorp/armeria/internal/common/util/ChannelUtil.java

```diff
@@ -45,6 +45,23 @@
 
 public final class ChannelUtil {
 
+    private static final String CHANNEL_PACKAGE_NAME;
+    @Nullable
+    private static final String INCUBATOR_CHANNEL_PACKAGE_NAME;
+
+    static {
+        // Determine the names of the Netty packages.
+        CHANNEL_PACKAGE_NAME = Channel.class.getPackage().getName();
+        final int lastDotIndex = CHANNEL_PACKAGE_NAME.lastIndexOf('.');
+        if (lastDotIndex > 0) {
+            // "shaded.io.netty.incubator.channel"
```

Comment:
Won't there be a dot and enter this branch even if not shaded?

Reply:
The result will be `io.netty.incubator.channel`, which is still correct. Am I missing out something?

Reply:
The comment seems confusing then since it seems to imply the branch is only for shaded :)

Reply:
Ah, OK. Let me update the comment. :laughing: